packages/pigeon/example/app/ios/Runner/EventChannelMessages.g.swift

```diff
@@ -23,6 +23,67 @@ private func nilOrValue<T>(_ value: Any?) -> T? {
   return value as! T?
 }
 
+func deepEqualsEventChannelMessages(_ lhs: Any?, _ rhs: Any?) -> Bool {
+  let cleanLhs = nilOrValue(lhs) as Any?
+  let cleanRhs = nilOrValue(rhs) as Any?
+  switch (cleanLhs, cleanRhs) {
+  case (nil, nil):
+    return true
+
+  case (nil, _), (_, nil):
+    return false
+
+  case is (Void, Void):
+    return true
+
+  case let (cleanLhsHashable, cleanRhsHashable) as (AnyHashable, AnyHashable):
+    return cleanLhsHashable == cleanRhsHashable
+
+  case let (cleanLhsArray, cleanRhsArray) as ([Any?], [Any?]):
+    guard cleanLhsArray.count == cleanRhsArray.count else { return false }
+    for (index, element) in cleanLhsArray.enumerated() {
+      if !deepEqualsEventChannelMessages(element, cleanRhsArray[index]) {
+        return false
+      }
+    }
+    return true
+
+  case let (cleanLhsDictionary, cleanRhsDictionary) as ([AnyHashable: Any?], [AnyHashable: Any?]):
+    guard cleanLhsDictionary.count == cleanRhsDictionary.count else { return false }
+    for (key, cleanLhsValue) in cleanLhsDictionary {
+      guard let cleanRhsValue = cleanRhsDictionary[key] else { return false }
+      if !deepEqualsEventChannelMessages(cleanLhsValue, cleanRhsValue) {
+        return false
+      }
+    }
+    return true
+
+  default:
+    return String(describing: cleanLhs) == String(describing: cleanRhs)
```

Comment:
where does the input come from? one could overwrite a type's `description` to return arbitrary strings.

Reply:
yeah, I couldn't find a better way to manage this besides just returning false. This seemed potentially better? This isn't (theoretically) hittable anyway since pigeon types are limited

Reply:
when will we hit the default case? we might as well just return `false` if we already covered all known cases that return `true`. Also worth adding a comment.